Kitodo/src/main/java/org/kitodo/production/services/file/FileService.java

```diff
@@ -102,6 +102,24 @@ URI createMetaDirectory(URI parentFolderUri, String directoryName) throws IOExce
         return URI.create(parentFolderUri.getPath() + '/' + directoryName);
     }
 
+    /**
+     * Creates a directory including any missing parent directories.
+     *
+     * @param pathToCreate
+     *            path to create
+     */
+    public void createDirectories(URI pathToCreate) throws IOException {
```

Comment:
please write a test for this.
I think you get an indexOutOfBounds exception if you pass a path without a slash here, which can happen if you iterate over and over and get to the last directory. (I see you expect the process directories to be existent, but it is a public method an anyone can pass an arbitrary folder to be created)

Reply:
Test added

Reply:
You didn't change the code here and you are still getting an index out of bounds exception.
You can not assume, that any part of the directory is already there!

Reply:
I do not know what I would have to change, because the method throws no exception for me. I even tried `new FileService().createDirectories(new File("F:\\create\\several\\directories\\").toURI());` on an freshly formatted USB stick, so there was **definitely nothing** there before, and did not throw an exception. What does the code look like which you used to get an exception there?